### DISCUSSION TITLE
Retention policy per API-key?

### DISCUSSION THREAD

**Discussion by Inbiten** [datalust/seq-tickets#1164] [Help]
Hi!
We have a situation where some data from specific services might have some GDPR in the logs.
We have one instance of seq where we would want to delete GDPR logs older than 20 days, while having another retention policy for all the other data. This way we can follow events through all services, up and until 20 days.
We thought I could be done by having different API-keys, but it seems that I was wrong.

Is there any way to solve this issue?

**Comment by nblumhardt** [comment]
Hi, thanks for getting in touch!

Different API keys should enable this. In the settings for an API key, you'll find a section called "Applied properties".

In there, for your GDPR-relevant logs, you can add a marker property e.g. `RetentionGroup` with the value `GDPR`.

You can then create a signal with the filter `RetentionGroup = 'GDPR'`,  and verify that it picks up the right logs.

The signal needs to be marked as "shared" from the drop-down in the signal editor beside its name. After that, you can create a retention policy to delete events matching that signal at 20 days.

Hope this helps!
